server/src/inputs/plugins/solarSystems.ts

```diff
@@ -0,0 +1,91 @@
+import SolarSystemPlugin from "server/src/classes/Plugins/Universe/SolarSystem";
+import {DataContext} from "server/src/utils/DataContext";
+import {pubsub} from "server/src/utils/pubsub";
+import {AstronomicalUnit, LightMinute} from "server/src/utils/unitTypes";
+import {getPlugin} from "./utils";
+
+export const solarSystemsPluginInputs = {
+  pluginSolarSystemCreate(
+    context: DataContext,
+    params: {
+      pluginId: string;
+      position: {
+        x: LightMinute;
+        y: LightMinute;
+        z: LightMinute;
+      };
+    }
+  ) {
+    const plugin = getPlugin(context, params.pluginId);
+    const solarSystem = new SolarSystemPlugin(
+      {position: params.position},
+      plugin
+    );
+    plugin.aspects.solarSystems.push(solarSystem);
+
+    pubsub.publish("pluginSolarSystems", {pluginId: params.pluginId});
+
+    return {solarSystemId: solarSystem.name};
+  },
+  async pluginSolarSystemDelete(
+    context: DataContext,
+    params: {
+      pluginId: string;
+      solarSystemId: string;
+    }
+  ) {
+    const plugin = getPlugin(context, params.pluginId);
+    const solarSystem = plugin.aspects.solarSystems.find(
+      solarSystem => solarSystem.name === params.solarSystemId
+    );
+    if (!solarSystem) return;
+    plugin.aspects.solarSystems.splice(
+      plugin.aspects.solarSystems.indexOf(solarSystem),
+      1
+    );
+
+    await solarSystem?.removeFile();
+    pubsub.publish("pluginShips", {pluginId: params.pluginId});
+  },
+  async pluginSolarSystemUpdate(
+    context: DataContext,
+    params: {
+      pluginId: string;
+      solarSystemId: string;
+      position?: {
+        x: LightMinute;
+        y: LightMinute;
+        z: LightMinute;
+      };
+      name?: string;
+      description?: string;
+      tags?: string[];
+      habitableZoneInner?: AstronomicalUnit;
+      habitableZoneOuter?: AstronomicalUnit;
+      skyboxKey?: string;
+    }
+  ) {
+    const plugin = getPlugin(context, params.pluginId);
+    if (!params.solarSystemId) throw new Error("Solar System ID is required");
+    const solarSystem = plugin.aspects.solarSystems.find(
+      solarSystem => solarSystem.name === params.solarSystemId
+    );
+    if (!solarSystem) return {solarSystemId: ""};
+    if (params.position) solarSystem.position = params.position;
```

Comment:
If these are all getting changed with the update, should solarSystem be a constant?

Reply:
In JavaScript, `const` just means you can reassign the variable, but you can still change the properties.

What I'm doing here is pulling out an object that's stored in the database and mutating the properties directly. So whether I use `const` or `let` doesn't really make a difference. It's just the way I prefer to write JavaScript.